code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/GithubOAuthImpl.java

```diff
@@ -36,31 +38,38 @@ public Account getAccount(ApiRequest request) {
                 return null;
             }
         }
-        String accountId = validateAndFetchAccountId(token);
-        return authDao.getAccountByExternalId(accountId, GITHUB_ACCOUNT_TYPE);
+        String accountId = githubUtils.validateAndFetchAccountIdFromToken(token);
+        if(null == accountId) {
+            return null;
+        }
+        Account account = authDao.getAccountByExternalId(accountId, GITHUB_ACCOUNT_TYPE);
+
+        String projectId = request.getServletContext().getRequest().getHeader(PROJECT_HEADER);
+        if (StringUtils.isEmpty(projectId)) {
+            return account;
+        }
+
+        String unobfuscatedId = ApiContext.getContext().getIdFormatter().parseId(projectId);
+        
+        Account projectAccount = authDao.getAccountById(new Long(unobfuscatedId));
 
+        if (validateProjectAccount(projectAccount, token)) {
+            return projectAccount;
+        }
+        return null;
     }
 
-    private String validateAndFetchAccountId(String token) {
-        String toParse = null;
-        String[] tokenArr = token.split("\\s+");
-        if (tokenArr.length == 2) {
-            if(!StringUtils.equals("bearer", StringUtils.lowerCase(StringUtils.trim(tokenArr[0])))) {
-                return null;
-            }
-            toParse = tokenArr[1];
-        } else if (tokenArr.length == 1) {
-            toParse = tokenArr[0];
-        } else {
-            return null;
+    protected boolean validateProjectAccount(Account projectAccount, String token) {
+        if (null == projectAccount) {
+            return false;
         }
-        Map<String, Object> jsonData;
-        try {
-            jsonData = tokenService.getJsonPayload(toParse, true);
-        } catch (TokenException e) { //in case of invalid token
-            return null;
+        Account account = projectAccount;
+        List<String> accesibleIds = githubUtils.validateAndFetchAccesibleIdsFromToken(token);
+        if (StringUtils.equals(USER_SCOPE, projectAccount.getExternalIdType())) {
+            account = authDao.getAccountById(new Long(projectAccount.getExternalId()));
```

Comment:
I don't understand this logic?  Why would externalId on the project equal a local account id?